pkg/util/kubernetes/apiserver/hpa_controller.go

```diff
@@ -39,34 +40,42 @@ type hpaToStoreGlobally struct {
 	data []custommetrics.ExternalMetricValue
 	m    sync.Mutex
 }
+
+// AutoscalersController is responsible for synchronizing objects from the Kubernetes
+// apiserver to build and cache the horizontal pod autoscalers in the cluster.
+// This allows to compute the External Metrics specified to autoscale pods and collect them from Datadog.
+//
+// The controller takes care to garbage collect any data while processing updates/deletes
+// so that the cache does not contain data for deleted hpas
+// This controller is used by the Datadog Cluster Agent and supports Kubernetes 1.10+.
 type AutoscalersController struct {
 	autoscalersLister       autoscalerslister.HorizontalPodAutoscalerLister
 	autoscalersListerSynced cache.InformerSynced
 	// Autoscalers that need to be added to the cache.
 	queue workqueue.RateLimitingInterface
 
-	// used in unit tests to wait until endpoints are synced
+	// used in unit tests to wait until hpas are synced
 	autoscalers chan interface{}
 
 	toStore   hpaToStoreGlobally
-	hpaProc   *hpa.HPAProcessor
+	hpaProc   *hpa.Processor
 	store     custommetrics.Store
 	clientSet kubernetes.Interface
-	poller    Poller
-	le        LeaderElectorItf
+	poller    PollerConfig
+	le        LeaderElectorInterface
 }
 
-// NewAutoscalerController returns a new AutoscalerController
-func NewAutoscalerController(client kubernetes.Interface, le LeaderElectorItf, dogCl hpa.DatadogClient, autoscalingInformer autoscalersinformer.HorizontalPodAutoscalerInformer) *AutoscalersController {
+// NewAutoscalersController returns a new AutoscalerController
```

Comment:
comment should say `AutoscalersControllers`

Reply:
`AutoscalersController`